controllers/replica_set_controller.go

```diff
@@ -559,6 +562,29 @@ func buildAutomationConfig(mdb mdbv1.MongoDBCommunity, auth automationconfig.Aut
 		Build()
 }
 
+func guessEnterprise(mdb mdbv1.MongoDBCommunity) bool {
+	overrideAssumption, err := strconv.ParseBool(os.Getenv(construct.MongoDBAssumeEnterpriseEnv))
```

Comment:
I don't understand this logic, where would we ever set this env var?

Reply:
So this is a backdoor that we might want to set at the Operator Deployment Spec level (in env. vars). It's not documented anywhere, it just prevents the operator from guessing and enforces the certain behavior. It's here just in case we do the guessing wrong.

Reply:
I see, that makes sense!

Reply:
Yes, definitely! I will be looking into the internal docs shortly!